tests/auth/BravaGuard.test.ts

```diff
@@ -0,0 +1,251 @@
+    });
+  });
+
+  describe('Transaction Validation', () => {
+    it('should allow transactions to sequence executor', async () => {
+      const fluidSupplyContract = await deploy(
+        'FluidSupply',
+        deployer,
+        await adminVault.getAddress(),
+        await logger.getAddress()
+      );
+      const fluidSupplyAddress = await fluidSupplyContract.getAddress();
+      await adminVault.proposeAction(getBytes4(fluidSupplyAddress), fluidSupplyAddress);
+      await adminVault.addAction(getBytes4(fluidSupplyAddress), fluidSupplyAddress);
+      await adminVault.proposePool('Fluid', tokenConfig.fUSDC.address);
+      await adminVault.addPool('Fluid', tokenConfig.fUSDC.address);
+
+      const token = 'USDC';
+      const amount = ethers.parseUnits('100', tokenConfig[token].decimals);
+      await fundAccountWithToken(await safe.getAddress(), token, amount);
+
+      const payload = await encodeAction({
+        type: 'FluidSupply',
+        amount,
+      });
+
+      await executeSequence(await safe.getAddress(), {
+        name: 'FluidSupplySequence',
+        callData: [payload],
+        actionIds: [getBytes4(fluidSupplyAddress)],
+      }, false);
+      
+    });
+
+    it('should allow pre-approved admin transactions', async () => {
+      const tx = {
+        to: await safe.getAddress(),
+        value: 0,
+        data: safe.interface.encodeFunctionData('setGuard', [ethers.ZeroAddress]),
+        operation: Operation.Call,
+        safeTxGas: 0,
+        baseGas: 0,
+        gasPrice: 0,
+        gasToken: ethers.ZeroAddress,
+        refundReceiver: ethers.ZeroAddress,
+      };
+
+      const signature = createSafeSignature(deployer);
+
+      // Should fail because this is not a pre-approved transaction
+      await expect(
+        safe.execTransaction(
+          tx.to,
+          tx.value,
+          tx.data,
+          tx.operation,
+          tx.safeTxGas,
+          tx.baseGas,
+          tx.gasPrice,
+          tx.gasToken,
+          tx.refundReceiver,
+          signature
+        )
+      ).to.be.revertedWithCustomError(bravaGuard, 'BravaGuard_TransactionNotAllowed');
+
+      const txHash = calculateTransactionHash(tx.to, tx.data, tx.operation);
+
+      // Propose and approve the transaction (testing delay is zero)
+      await transactionRegistry.proposeTransaction(txHash);
+      await transactionRegistry.approveTransaction(txHash);
+
+      const isApproved = await transactionRegistry.isApprovedTransaction(txHash);
+      expect(isApproved).to.be.true;
+
+
+      // Should now succeed because the transaction is pre-approved
+      await expect(
+        safe.execTransaction(
+          tx.to,
+          tx.value,
+          tx.data,
+          tx.operation,
+          tx.safeTxGas,
+          tx.baseGas,
+          tx.gasPrice,
+          tx.gasToken,
+          tx.refundReceiver,
+          signature
+        )
+      ).to.not.be.reverted;
```

Comment:
Should we also check here if the guard was actually removed? I'm used to getting a lot of false positives with Safe.